### DISCUSSION TITLE
QT 6.3 / Visual Studio 2022 Community / Git - help&diskussion thread

### DISCUSSION THREAD

**Discussion by h3rm** [emericg/WatchFlower#62] [General]
Hi!

Like @emericg wrote i will open a discussion for building the watchflower.git on windows (11)-PC's.

Until now i found out:
1. https://www.qt.io/download-qt-installer show 6.2. in the installer?_ -> if you select "custom installation" -> voila - 6.3. beta :)
2. Visual Studio 2022 Community (or maybe Visual Studio Code also works?)
3. Python (unsure if standalone or via visual studio internally)

> 4. (unsure) https://emscripten.org/docs/getting_started/downloads.html#installation-instructions-using-the-emsdk-recommended) to install with python. Also the other required tools (https://emscripten.org/docs/building_from_source/toolchain_what_is_needed.html#toolchain-what-you-need)  

Note "4": ignore the error in QT, no need for that as emeric wrote

> [emericg](https://github.com/emericg) commented [9 hours ago](https://github.com/emericg/WatchFlower/issues/57#issuecomment-1080441791)
> Otherwise, with Qt 6.3 and up you actually need Visual Studio on Windows, otherwise the Bluetooth won't work. I think both 2019 and 2022 are fine (community edition is good too).
> Because WatchFlower has no external dependencies, the process of building it is straightforward.
> Install MSVC, install Qt 6.3 with the Qt installer (don't forget Bluetooth and other modules).
> Load the WatchFlower.pro file with Qt Creator (MSVC and Qt 6 should be autodetected) then build :)

Yesterday i tried around with windows 11 a lot of things (QT/Git/VS 2022/VS code/Python) but in QT 6.3 i couldn't solve a compiler problem in the webassembly-settings because i wasn't able to install https://emscripten.org/docs/getting_started/downloads.html#installation-instructions-using-the-emsdk-recommended) with python. Also the other required tools (https://emscripten.org/docs/building_from_source/toolchain_what_is_needed.html#toolchain-what-you-need) are strange to install and to use in Windows...  Maybe i will startover again with linux soon and use the console instead for everything:) 

![PiyGiWT](https://user-images.githubusercontent.com/9608955/160471879-4eaee33c-d64b-4401-bb26-29d854f4ba02.png)

![5qlPawZ](https://user-images.githubusercontent.com/9608955/160471883-78cc86cb-25f7-4b0c-8223-0f8dfbec759e.png)

![Swg9xTM](https://user-images.githubusercontent.com/9608955/160471884-11340c4b-3ab0-4bd4-8e28-7740a8737138.png)

![84GGIad](https://user-images.githubusercontent.com/9608955/160471887-7c349b36-b183-4307-a104-a4a7d1e01411.png)

![4KAbAH2](https://user-images.githubusercontent.com/9608955/160471888-03d20948-dc85-4928-ab64-ae1b661e795e.png)

As you can see i can't add a device at the moment because i get the message that i  have to install https://emscripten.org/docs/getting_started/downloads.html#installation-instructions-using-the-emsdk-recommended . but it seems there are more problems in the Kits/QT Version/Compiler settings.. a fresh install of QT didn't work because the settings can't be uninstalled.

Maybe i startover again on linux because i don't really understand python for windows - nearly all command don't work in the python.exe ...

> (Please create a discussion anyway so we can continue this conversation ^^)
> 
> I mean you really don't need neither python nor webassembly... And you certainly don't need visual studio code.
> I don't know much about VS Code but as far as I am concerned it's a text editor, it has nothing to do with compiling WatchFlower.
> 
> If you are not really proficient with any of the software involved, you really should stick to Qt Creator.
> 
> Again, you only need :
> 
>     MSVC 2019 (community edition is ok). That's the real visual studio, not the text editor.
>     Qt 6.3 with the modules, installed with the Qt Maintenance Tool, very convenient.
>     Qt Creator, also installed by default with the maintenance tool.
> 
> Then open WatchFlower.pro with Qt Creator and press the play button...

I learned basics in Perl, C++ and ij Unreal Engine and Blender.. i know the basics and understand the workflows.. but yesterday i gave up because the 5 tools made me crazy :) i will start over on friday with your tutorial :) thanks for that!

**Comment by emericg** [comment]
A "kit" is a target for the application.
- You have kits for Android, one for each hardware version of Android (chromebooks don't have the same processors than smartphones, and old smartphones don't have the same processors than today's smartphone, that's why there is 4 kits). Also, Qt/Android on Windows is NOT an entry level setup). At all.
- You have two kits for windows, 32 and 64b, that's the kit you want to use.
- There is one for Windows ARM. You probably have no need for that.
- The Emscipten kit is... You just don't want to touch that until you reaaaaaaalllllllly know what you are doing... And then, probably not either.

When you load a project file (like WatchFlower.pro) with Qt creator it will ask you what kit you want to use. You should choose the "MSVC2019 64bit" one.

And again yes you don't need python (maybe one is installed by default but it's not important for your use case), VS Code or emscripten.

**Reply by h3rm** [reply]
![2022-03-29 21_07_06-Einstellungen](https://user-images.githubusercontent.com/9608955/160687979-417a6603-8b34-4a93-a718-d2b99c3f78d6.png)

after nearly two days playing around with the QT + MSVS2022-settings, understanding both interfaces and their settings and 5-6 uninstalls/reinstalls (plus 10 python-installs....) i finally got it! thx mate! 🙏 👍 🥇 

next step: understanding your code and find out how QT itself, design adjustments + github(lol) works ..... 🤣

**Reply by emericg** [reply]
That's good to hear!
Let me know if you have an area of interest or features you want to work on. There are a lot of stuff that I'm already working on that you can't find yet in public versions.

**Reply by h3rm** [reply]
> That's good to hear! Let me know if you have an area of interest or features you want to work on. There are a lot of stuff that I'm already working on that you can't find yet in public versions.

it wil take some time until i learned everything i need but i will contact you as soon as i can help to improve your great project 🤝
